src/tasks/block.rs

```diff
@@ -139,39 +141,104 @@ impl BlockBuilder {
         }
     }
 
+    /// Fetches transactions from the cache and ingests them into the in progress block
     async fn get_transactions(&mut self, in_progress: &mut InProgressBlock) {
-        tracing::trace!("query transactions from cache");
+        trace!("query transactions from cache");
         let txns = self.tx_poller.check_tx_cache().await;
         match txns {
             Ok(txns) => {
-                tracing::trace!("got transactions response");
+                trace!("got transactions response");
                 for txn in txns.into_iter() {
                     in_progress.ingest_tx(&txn);
                 }
             }
             Err(e) => {
-                tracing::error!(error = %e, "error polling transactions");
+                error!(error = %e, "error polling transactions");
             }
         }
     }
 
-    async fn _get_bundles(&mut self, in_progress: &mut InProgressBlock) {
-        tracing::trace!("query bundles from cache");
+    /// Fetches bundles from the cache and ingests them into the in progress block
+    async fn get_bundles(
+        &mut self,
+        ru_provider: &WalletlessProvider,
+        in_progress: &mut InProgressBlock,
+    ) {
+        // Authenticate before fetching to ensure access to a valid token
+        if let Err(err) = self.bundle_poller.authenticator.authenticate().await {
+            error!(err = %err, "bundle fetcher failed to authenticate");
+            return;
+        }
+
+        trace!("query bundles from cache");
         let bundles = self.bundle_poller.check_bundle_cache().await;
+        // TODO: Sort bundles received from cache
         match bundles {
             Ok(bundles) => {
-                tracing::trace!("got bundles response");
                 for bundle in bundles {
-                    in_progress.ingest_bundle(bundle);
+                    let result = self.simulate_bundle(&bundle.bundle, ru_provider).await;
```

Comment:
alright, so as we talked yday I think let's just handwave simulation away from now. Once @prestwich has the sim engine ready, we can plug that in, whenever that happens. I wouldn't worry about this at all for now as it's introducing multiple caveats when simulating through `call`.